lib/Jsrt/JsrtDebuggerObject.cpp

```diff
@@ -808,43 +822,51 @@ Js::JavascriptArray * JsrtDebugStackFrames::StackFrames(Js::ScriptContext * scri
         this->framesDictionary->Clear();
     }
 
-    uint frameCount = 0;
-
     ThreadContext* threadContext = scriptContext->GetThreadContext();
 
     DWORD_PTR dispatchHaltFrameAddress = threadContext->GetDebugManager()->GetDispatchHaltFrameAddress();
     AssertMsg(dispatchHaltFrameAddress, "Didn't set the dispatchHaltFrameAddress at time of break?");
 
-    for (Js::ScriptContext *tempScriptContext = threadContext->GetScriptContextList();
-    tempScriptContext != nullptr && tempScriptContext->IsScriptContextInDebugMode();
-        tempScriptContext = tempScriptContext->next)
+    if (dispatchHaltFrameAddress != 0)
```

Comment:
fair enough. Could you change above assert expression to dispatchHaltFrameAddress > 0, just to make it more readable? Also framesDictionary can be moved under this check in that case.

---
In reply to: [68853053](https://github.com/Microsoft/ChakraCore/pull/1204#discussion_r68853053) [](ancestors = 68853053,68851362)